packages/amplify-cli/src/extensions/amplify-helpers/open-editor.ts

```diff
@@ -19,47 +21,89 @@ export async function openEditor(context, filePath, waitToContinue = true) {
 
   if (editorSelected !== 'none') {
     const editorArguments: string[] = [];
-    const editor = envEditor.getEditor(editorSelected);
+
+    let editor: envEditor.Editor;
+
+    // Until env-editor supports xcode we've to have a branch here
+    if (editorSelected === 'xcode') {
+      editor = {
+        id: 'xcode',
+        name: 'Xcode',
+        binary: 'xcode',
+        isTerminalEditor: false,
+        paths: ['/Applications/Xcode.app/Contents/MacOS/Xcode', '/Applications/Xcode-beta.app/Contents/MacOS/Xcode'],
+        keywords: [],
+      };
+    } else {
+      editor = envEditor.getEditor(editorSelected);
+    }
 
     if (!editor) {
-      console.error(
+      context.print.error(
         `Selected editor '${editorSelected}' was not found in your machine. Please open your favorite editor and modify the file if needed.`,
       );
     }
-    const editorPath = editor.paths.find(p => fs.existsSync(p));
 
-    if (editorSelected === 'vscode') {
-      editorArguments.push('--goto');
+    let editorPath: string | undefined = editor.paths.find(p => fs.existsSync(p));
+
+    // Check if the binary can be located with which
+    if (!editorPath) {
+      const resolvedBinary = which.sync(editor.binary, {
+        nothrow: true,
+      });
+
+      if (resolvedBinary !== null) {
+        editorPath = resolvedBinary;
+      }
     }
 
-    editorArguments.push(filePath);
+    // In case if the selected editor was found.
+    if (!editorPath) {
+      context.print.warning(`Couldn’t find selected code editor (${editorSelected}) on your machine.`);
 
-    try {
-      if (!editor.isTerminalEditor) {
-        const subProcess = execa(editorPath || editor.binary, editorArguments, {
-          detached: true,
-          stdio: 'ignore',
-        });
+      const openFile = await context.amplify.confirmPrompt('Try opening with system-default editor instead?', true);
```

Comment:
should we handle the `--yes` flag here?

Reply:
This is a "downstream" question that in case of `--yes` open editor itself should not be called at all. So in this particular case I was testing, opening the GraphQL file for edit it is something that the API category question should handle.